packages/discord.js/src/structures/MessageMentions.js

```diff
@@ -184,17 +184,21 @@ class MessageMentions {
    * @returns {boolean}
    */
   has(data, { ignoreDirect = false, ignoreRoles = false, ignoreEveryone = false } = {}) {
-    if (!ignoreEveryone && this.everyone) return true;
-    const { GuildMember } = require('./GuildMember');
-    if (!ignoreRoles && data instanceof GuildMember) {
-      for (const role of this.roles.values()) if (data.roles.cache.has(role.id)) return true;
-    }
+    const user = this.client.users.resolve(data);
+    const role = this.guild?.roles.resolve(data);
+    const channel = this.client.channels.resolve(data);
 
     if (!ignoreDirect) {
-      const id =
-        this.guild?.roles.resolveId(data) ?? this.client.channels.resolveId(data) ?? this.client.users.resolveId(data);
-
-      return typeof id === 'string' && (this.users.has(id) || this.channels.has(id) || this.roles.has(id));
+      if (this.users.has(user?.id)) return true;
+      if (!ignoreRoles && this.roles.has(role?.id)) return true;
```

Comment:
```suggestion
      if (this.roles.has(role?.id)) return true;
```
Should ignoreRole be only in member check?